src/puptoo/upload.py

```diff
@@ -0,0 +1,35 @@
+import io
+import json
+from minio import Minio
+from .utils import config, puptoo_logging
+
+logger = puptoo_logging.initialize_logging()
+
+
+# Minio client setup and upload yum_updates object
+def upload_object(yum_updates, extra, msg):
+    _bytes = json.dumps(yum_updates).encode("utf-8")
+    client = Minio(endpoint=config.S3_ENDPOINT, access_key=config.ACCESS_KEY, secret_key=config.SECRET_KEY, secure=config.USE_SSL)
+    if client.bucket_exists(config.BUCKET_NAME):
+        try:
+            client.put_object(bucket_name=config.BUCKET_NAME, object_name=extra["request_id"], data=io.BytesIO(_bytes), length=len(_bytes))
+            logger.info(
+                "Successfully uploaded object (%s) to s3 bucket", extra["request_id"]
+            )
+        except:
+            logger.exception("An error occurred while uploading object")
+    else:
+        logger.error("Bucket (%s) does not exist", config.BUCKET_NAME)
+        
+    msg["custom_metadata"]["yum_updates_s3url"] = get_url(client, extra["request_id"])
```

Comment:
Can we move this line 24 inside the try block(on line 19). Currently if puptoo is not able to upload archive to minio it still creates the invalid `yum_updates_s3url` link and forwards to host-ingress topic. If we try to wget this link it returns 404 error.